tests/test_xpartamupp.py

```diff
@@ -70,20 +70,99 @@ def test_remove(self):
+        jids.append(JID(jid="player4@domain.tld", resource="0ad-bnmp6789"))
+
+        game_data1 = {'players': ['player1', 'player2'], 'nbp': 'foo', 'state': 'init'}
+        games.add_game(jids[0], game_data1)
+        game_data2 = {'players': ['player3', 'player4'], 'nbp': 'bar', 'state': 'init'}
+        games.add_game(jids[1], game_data2)
+        game_data3 = {'players': ['player5', 'player6'], 'nbp': 'bar', 'state': 'init'}
+        games.add_game(jids[2], game_data3)
+        game_data4 = {'players': ['player7', 'player8'], 'nbp': 'foo', 'state': 'init'}
+        games.add_game(jids[3], game_data4)
+        game_data5 = {'players': ['player9', 'playera'], 'nbp': 'bar', 'state': 'init'}
+        games.add_game(jids[4], game_data5)
+        game_data6 = {'players': ['playerb', 'playerc'], 'nbp': 'bar', 'state': 'init'}
+        games.add_game(jids[5], game_data6)
+
+        game_data1.update({'players-init': game_data1['players'],
+                           'nbp-init': game_data1['nbp'],
+                           'state': game_data1['state']})
+        game_data2.update({'players-init': game_data2['players'],
+                           'nbp-init': game_data2['nbp'],
+                           'state': game_data2['state']})
+        game_data3.update({'players-init': game_data3['players'],
+                           'nbp-init': game_data3['nbp'],
+                           'state': game_data3['state']})
+        game_data4.update({'players-init': game_data4['players'],
+                           'nbp-init': game_data4['nbp'],
+                           'state': game_data4['state']})
+        game_data5.update({'players-init': game_data5['players'],
+                           'nbp-init': game_data5['nbp'],
+                           'state': game_data5['state']})
+        game_data6.update({'players-init': game_data6['players'],
+                           'nbp-init': game_data6['nbp'],
+                           'state': game_data6['state']})
+
+        expected_games_dict = {
+            jids[0].bare: {jids[0].resource: game_data1, jids[1].resource: game_data2,
+                           jids[2].resource: game_data3},
+            jids[3].bare: {jids[3].resource: game_data4},
+            jids[4].bare: {jids[4].resource: game_data5},
+            jids[5].bare: {jids[5].resource: game_data6}}
+        self.assertDictEqual(expected_games_dict, games.games)
+
+        expected_games_list = [game
+                               for jid, dict_of_resources_and_games
+                               in sorted(expected_games_dict.items(),
+                                         key=lambda item: item[0])      # sort by jid
+                               for resource, game
+                               in sorted(dict_of_resources_and_games.items(),
+                                         key=lambda item: item[0])]     # sort by resource
+        self.assertListEqual(games.get_all_games(), expected_games_list)
```

Comment:
You test the functionality of `Games.get_all_games()` here by duplicating its implementation. How is this useful and how does it relate to testing if removing all game for a jid works as expected?